cms/lib/xblock/upstream_sync.py

```diff
@@ -81,18 +81,72 @@ class UpstreamLink:
     version_available: int | None  # Latest version of the upstream that's available, or None if it couldn't be loaded.
     version_declined: int | None  # Latest version which the user has declined to sync with, if any.
     error_message: str | None  # If link is valid, None. Otherwise, a localized, human-friendly error message.
+    has_top_level_parent: bool  # True if this Upstream link has a top-level parent
+    ready_to_sync: bool  # Should we invite the downstream's authors to sync the latest upstream updates?
 
-    @property
-    def ready_to_sync(self) -> bool:
+    def calculate_ready_to_sync(self, downstream) -> bool:
         """
-        Should we invite the downstream's authors to sync the latest upstream updates?
+        Calculates the ready to sync value using the version available.
+        If is a container, also verifies if the children needs sync.
         """
-        return bool(
-            self.upstream_ref and
-            self.version_available and
-            self.version_available > (self.version_synced or 0) and
-            self.version_available > (self.version_declined or 0)
-        )
+        if isinstance(self.upstream_key, LibraryUsageLocatorV2):
+            return bool(
+                self.upstream_ref and
+                self.version_available and
+                (not self.has_top_level_parent) and  # <- If has top-level parent, we need to sync the parent
+                self.version_available > (self.version_synced or 0) and
+                self.version_available > (self.version_declined or 0)
+            )
+        elif isinstance(self.upstream_key, LibraryContainerLocator):
+            # If this container has top-level parent, so we need to sync the parent
+            if self.has_top_level_parent:
+                return False
+
+            container_ready_to_sync = bool(
+                self.upstream_ref and
+                self.version_available and
+                self.version_available > (self.version_synced or 0) and
+                self.version_available > (self.version_declined or 0)
+            )
```

Comment:
We can move the common simple check up.
```suggestion
        # If this container or block has top-level parent, we need to sync the parent
        if self.has_top_level_parent:
            return False

        if isinstance(self.upstream_key, LibraryUsageLocatorV2):
            return bool(
                self.upstream_ref and
                self.version_available and
                (not self.has_top_level_parent) and  # <- If has top-level parent, we need to sync the parent
                self.version_available > (self.version_synced or 0) and
                self.version_available > (self.version_declined or 0)
            )
        elif isinstance(self.upstream_key, LibraryContainerLocator):
            container_ready_to_sync = bool(
                self.upstream_ref and
                self.version_available and
                self.version_available > (self.version_synced or 0) and
                self.version_available > (self.version_declined or 0)
            )
```

Reply:
Updated: https://github.com/openedx/edx-platform/pull/37217/commits/c2589b4828073d13418953f62bc345bb6deb67b8